### PR TITLE
wasmate: trim zeros from both ends of the data segment.

### PR DESCRIPTION
**Pull request by ncbray**

### CODE REVIEW

**Comment by dschuff** [comment]
In general I like the refactoring. What's the purpose of  representing the zeros specially though? Also I'm not sure I like using non-zero values for parts to-be-relocated.

**Comment by ncbray** [comment]
Added comments.  The purpose is to reduce the size of the raytracer example from 262k => 1k.  It statically allocates a frame buffer in a global variable, which in turn dumps a bunch of zeros into the data segment.

**Comment by dschuff** [comment]
LGTM. I suppose it's only a matter of time before we implement BSS :/

File: prototype-wasmate/wasmate.py [hunk 1]
```diff
@@ -157,36 +157,81 @@ def reduce_to_bytes(x, num_bytes):
...
+    def align_to(self, align):
+        while self.end() % align != 0:
+            self.trailing_zeros += 1
+
+    def fixup(self, addr, value, num_bytes):
+        pos = addr - self.base
+        b = reduce_to_bytes(value, num_bytes)
+        self.data[pos:pos + num_bytes] = b
+
+    def appendByte(self, byte):
```
Comment (binji) [inline prototype-wasmate/wasmate.py]: file style seems to be append_byte

**Pull request merged**